src/store/groups/reducers.ts

```diff
@@ -1,62 +1,110 @@
-import { 
-  GroupsState, 
-  GroupsActionTypes, 
-  FETCH_GROUP_TABLE_SUCCESS, 
-  Group, 
-  FETCH_GROUP_TABLE, 
-  SEARCH_GROUP_TABLE, 
-  FETCH_GROUP_TABLE_FAILURE 
-} from "./types";
-
-const initialState: GroupsState = {
-    groups: [],
-    tableView: {
-      selectedGroupId: 0,
-      table: []
-    },
-    fetching: false,
-    hasFetchingError: false,
-    searchText: ''
-}
+import {
+	GroupsState,
+	CreateGroupState,
+	GroupsActionTypes,
+	CreateGroupActionTypes,
+	FETCH_GROUP_TABLE_SUCCESS,
+	FETCH_GROUP_TABLE,
+	SEARCH_GROUP_TABLE,
+	FETCH_GROUP_TABLE_FAILURE,
+	CREATE_GROUP,
+	CREATE_GROUP_SUCCESS,
+	CREATE_GROUP_FAILURE,
+	RESET_CREATE_GROUP_STATE,
+} from './types';
+
+const groupInitialState: GroupsState = {
+	groups: [],
+	tableView: {
+		selectedGroupId: 0,
+		table: [],
+	},
+	fetching: false,
+	hasFetchingError: false,
+	searchText: '',
+};
 
 export function groupReducer(
-    state = initialState,
-    action: GroupsActionTypes
-  ): GroupsState {
-    switch (action.type) {
-      case FETCH_GROUP_TABLE:
-        return {
-          ...state,
-          tableView: {
-            selectedGroupId: action.payload,
-            table: []
-          },
-          fetching: true,
-          hasFetchingError: false
-        }
-      case FETCH_GROUP_TABLE_SUCCESS:
-        return {
-          ...state,
-          tableView: {
-            selectedGroupId: action.payload.groupId,
-            table: action.payload.table
-          },
-          fetching: false,
-          hasFetchingError: false,
-        }
-      case FETCH_GROUP_TABLE_FAILURE:
-        return {
-          ...state,
-          fetching: false,
-          hasFetchingError: true,
-        }
-      case SEARCH_GROUP_TABLE:
-          return {
-            ...state,
-            searchText: action.payload,
-            hasFetchingError: false,
-          }
-      default:
-        return state
-    }
-  }
\ No newline at end of file
+	state = groupInitialState,
+	action: GroupsActionTypes
+): GroupsState {
+	switch (action.type) {
+		case FETCH_GROUP_TABLE:
+			return {
+				...state,
+				tableView: {
+					selectedGroupId: action.payload,
+					table: [],
+				},
+				fetching: true,
+				hasFetchingError: false,
+			};
+		case FETCH_GROUP_TABLE_SUCCESS:
+			return {
+				...state,
+				tableView: {
+					selectedGroupId: action.payload.groupId,
+					table: action.payload.table,
+				},
+				fetching: false,
+				hasFetchingError: false,
+			};
+		case FETCH_GROUP_TABLE_FAILURE:
+			return {
+				...state,
+				fetching: false,
+				hasFetchingError: true,
+			};
+		case SEARCH_GROUP_TABLE:
+			return {
+				...state,
+				searchText: action.payload,
+				hasFetchingError: false,
+			};
+		default:
+			return state;
+	}
+}
+
+const createGroupInitialState: CreateGroupState = {
+	submitted: false,
+	submitting: false,
+	hasError: false,
+	groupId: '',
+};
+
+export function createGroupReducer(
```

Comment:
i think there should be one group reducer and these actions should be in it it should be in the grou